active/gridsyntax/GridSyntaxSpec2.md

```diff
@@ -0,0 +1,368 @@
+
+But the Xaml syntax rules now support this scenario by recognizing that the target property is a collection type, and that there's no conversion from string available, and then by treating the attribute value as a comma-separated list of collection items. The items of the collection are of type [ColumnDefinition](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Controls.ColumnDefinition), so a check is made to see if strings can be converted to ColumnDefinition instances. Since they can, the Xaml loader creates those instances and adds them to the collection.
+
+| Do we have documentation that explains what exactly a "collection" is? Can't remember if it's just IVector, or if it has more rules like the C# initialization syntax.
+
+## Role of the [ContentProperty] attribute
+
+In order for a ColumnDefinition (or RowDefinition) to be created from a string there is a second new Xaml syntax rule: if a type can't be created from string, but the type's [[ContentProperty](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Markup.ContentPropertyAttribute)] property can, the type is created and the property set.
+
+About this "content property", it's not literally a property named "content" (although it can be and [is sometimes](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Controls.ContentControl.Content)). This is essentially the default property of a class, and is marked with the [ContentProperty] attribute.
+
+For example the [SolidColorBrush](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Media.SolidColorBrush) is defined roughly like this (in C# terms):
+
+```cs
+[ContentProperty(Name = "Color")]
+public sealed class SolidColorBrush : Brush
+{
+    ...
+    public Color Color { get; set; }
+}
+```
+
+Because [SolidColorBrush.Color](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Media.SolidColorBrush.Color) is marked as the [ContentProperty] (and because a Color can be created from a string), you've always been able to write this short-hand Xaml:
+
+```xml
+<SolidColorBrush>Red</SolidColorBrush>
+```
+
+If it weren't for the [ContentProperty] on SolidColorBrush, you'd have to use the equivalent full syntax (and that's still valid to use if you prefer):
+
+```xml
+<SolidColorBrush>
+    <SolidColorBrush.Color>
+        Red
+    </SolidColorBrush.Color>
+</SolidColorBrush>
+```
+
+## Making things work for Grid
+
+Back to Grid, with the new syntax feature and with ColumnDefinition.Width now marked as the [ContentProperty]:
+
+```cs
+[ContentProperty(Name = "Width")]
+public sealed class ColumnDefinition : DependencyObject
+{
+        public GridLength Width { get; set; }
+}
+```
+
+you can now write
+
+```xml
+<ColumnDefinition>Auto</ColumnDefinition>
+```
+
+(There's already a converter to create a GridLength from a string.)
+
+Putting that all together, back to the original goal:
+
+<Grid ColumnDefinitions="Auto, *">
```

Comment:
make markdown?

Reply:
Oops, it still colors correctly in VSCode :)